src/posix/platform/system.cpp

```diff
@@ -196,6 +192,61 @@ void platformSetUp(void)
     return;
 }
 
+void otSysBorderRouterInit(otPlatformConfig *aPlatformConfig, int aIcmp6Socket)
+{
+    OT_UNUSED_VARIABLE(aPlatformConfig);
+    OT_UNUSED_VARIABLE(aIcmp6Socket);
+
+#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
+    ot::Posix::InfraNetif::Get().Init(aPlatformConfig->mBackboneInterfaceName, aIcmp6Socket);
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    platformTrelInit(getTrelRadioUrl(aPlatformConfig));
+#endif
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    platformBackboneInit(aPlatformConfig->mBackboneInterfaceName);
+#endif
+}
+
+void otSysBorderRouterSetUp(void)
+{
+#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
+    ot::Posix::InfraNetif::Get().SetUp();
+#endif
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    platformBackboneSetUp();
+#endif
+}
+
+void otSysBorderRouterTearDown(void)
+{
+#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
+    ot::Posix::InfraNetif::Get().TearDown();
+#endif
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    platformBackboneTearDown();
+#endif
+}
+
+void otSysBorderRouterDeinit(void)
+{
+#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
+    ot::Posix::InfraNetif::Get().Deinit();
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    platformTrelDeinit();
+#endif
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    platformBackboneDeinit();
+#endif
```

Comment:
nit: do the steps in the reverse order of them in `otSysBorderRouterInit`.

Reply:
I think it's unnecessary to do them in the reverse order because these features are not depending on each other.